src/pbkdf2.rs

```diff
@@ -133,12 +133,14 @@ use {constant_time, digest, hmac, polyfill};
 ///
 /// `derive` panics if `iterations < 1`.
 ///
-/// `derive` panics if `out.len()` is larger than the output length of the
-/// digest function used by the PRF algorithm.
+/// `derive` panics if `out.len()` is larger than 2^32 - 1, per the
+/// PBKDF2 specification
 pub fn derive(prf: &'static PRF, iterations: usize, salt: &[u8], secret: &[u8],
               out: &mut [u8]) {
     assert!(iterations >= 1);
-    assert!(out.len() <= prf.digest_alg.output_len);
+    assert!(out.len() <= 0xFFFFFFFF);
```

Comment:
The spec says `0xffffffff * output_len, not just 0xffffffff`.

Note that this check is only necessary and only possible to do when usize is larger than 32-bits. Otherwise, `0xffffffff * output_len` itself would overflow.

It seems good to keep the limit at `0xffffffff` to avoid the messiness, but at least the doc comment is wrong and needs to be updated.